tests/internal/parser.c

```diff
@@ -123,6 +128,35 @@ int flb_parser_regex_do(struct flb_parser *parser,
                         void **out_buf, size_t *out_size,
                         struct flb_time *out_time);
 
+static char* mock_timezone(char *tz)
```

Comment:
I think it would be nicer if `mock_timezone` returned an int with success, and took a pointer to where the original timezone should be returned. Then instead of doing `perror` and `exit` here, the test can see a `-1` from this function and go through the existing `acutest` failure pattern.